database.hh

```diff
@@ -312,9 +312,35 @@ struct cf_stats {
 
 class table;
 using column_family = table;
+struct table_stats;
+using column_family_stats = table_stats;
 
 class database_sstable_write_monitor;
 
+struct table_stats {
+    /** Number of times flush has resulted in the memtable being switched out. */
+    int64_t memtable_switch_count = 0;
+    /** Estimated number of tasks pending for this column family */
+    int64_t pending_flushes = 0;
+    int64_t live_disk_space_used = 0;
+    int64_t total_disk_space_used = 0;
+    int64_t live_sstable_count = 0;
+    /** Estimated number of compactions pending for this column family */
+    int64_t pending_compactions = 0;
+    int64_t partition_creations = 0;
+    int64_t partition_reuses = 0;
+    int64_t row_creations = 0;
+    int64_t row_overwrites = 0;
+    utils::timed_rate_moving_average_and_histogram reads{256};
+    utils::timed_rate_moving_average_and_histogram writes{256};
+    utils::estimated_histogram estimated_read;
+    utils::estimated_histogram estimated_write;
+    utils::estimated_histogram estimated_sstable_per_read{35};
+    utils::timed_rate_moving_average_and_histogram tombstone_scanned;
+    utils::timed_rate_moving_average_and_histogram live_scanned;
+    utils::estimated_histogram estimated_coordinator_read;
+};
+
```

Comment:
Why not to have cheap stats always enabled while expensive (histograms) conditionally enabled?

This would also resolve that Avi's comment about the conditional stats update and would allow you drop that workaround that you made.

Reply:
@vladzcloudius  If I understand correctly that you mean splitting the struct and adding a required argument to `memtable`'s constructor, then it would generate a big change:
```
[piodul@localhost scylla]$ git grep "make_lw_shared<memtable>" | wc -l
180
```
Almost all of those would need to be modified to include the new dummy parameter.
Furthermore, only one of those invocations is in non-test code and it always is passed a real argument, the rest is the test code.